### ISSUE TITLE
git crashes when reading desi data

### ISSUE THREAD

**Issue by ngbusca** [igmhub/picca#346]
@Cyeche reported:
```
bin/do_deltas.py --in-dir Spectra_desi/ --drq cat_QSO_DESI.fits --out-dir ./ --mode desi --nspec 20
```
Cela donne :
```
global/u2/y/yeche/Packages/lib/python2.7/site-packages/pyfits-3.5-py2.7-linux-x86_64.egg/pyfits/__init__.py:22: PyFITSDeprecationWarning: PyFITS is deprecated, please use astropy.io.fits
  PyFITSDeprecationWarning)  # noqa
 zqso_min = 2.0
 zqso_max = 4.28846153846
mode: desi start               : nb object in cat = 172162
 and thid>0          : nb object in cat = 172162
 and ra!=dec         : nb object in cat = 172162
 and ra!=0.          : nb object in cat = 172162
 and dec!=0.         : nb object in cat = 172162
 and z>0.            : nb object in cat = 172162
 and z>zmin          : nb object in cat = 48473
 and z<zmax          : nb object in cat = 48473
BAL_FLAG_VI not found
Found 20 qsos
read 0 of 1. ndata: 0found 20 quasars in input files
Traceback (most recent call last):
  File "bin/do_deltas.py", line 192, in <module>
    best_obs=args.best_obs, single_exp=args.single_exp, pk1d=args.delta_format )
ValueError: need more than 3 values to unpack
```

**Comment by londumas** [comment]
To keep track of the email discussions, the solution should be something like this:
```
result = read_from_desi(nside,in_dir,thid,ra,dec,zqso,plate,mjd,fid,order)
return result[0], result[1], nside,"RING"
```

**Comment by ngbusca** [comment]
I'm working on this. In my fix, read_from_desi won't return ndata, so it has the same behavior as the other read_from_X functions. I'll PR as soon as I can try it on desi (but I need a fix of #347  so I can do that)

**Comment by londumas** [comment]
@ngbusca, to fix https://github.com/igmhub/picca/issues/347 I need desi qso catalogs
